sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -213,8 +215,16 @@ private[hive] trait HiveInspectors {
 
     case c: Class[_] if c.isArray => ArrayType(javaClassToDataType(c.getComponentType))
 
+    // list type
+    case c: Class[_] if c == classOf[java.util.List[java.lang.Object]] =>
+      logWarning("Failed to catch a correct component type in List<> because of type erasure," +
+        " so you need to handle it correctly by yourself")
```

Comment:
I'm not sure the best one as this warning message.
Is " so you need to cast it into the correct type by yourself" ok?